src/coreclr/src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -90,7 +104,7 @@ internal object SyncRoot
         // is returned.
         public override AssemblyName GetName(bool copiedName)
         {
-            string? codeBase = GetCodeBase(copiedName);
+            string? codeBase = GetCodeBase();
```

Comment:
Does this mean that if I have an `Assembly asm` loaded from the bundle, `asm.CodeBase` will throw, but `asm.GetName().CodeBase` will be `null`?

Reply:
Yup. I couldn't see a good way of making the behavior match. I don't think there's anything wrong with fetching an AssemblyName, so I don't think that should throw, but since the AssemblyName doesn't have a pointer back to the Assembly there's no way to hook just the CodeBase API, aside from doing something drastic like subclassing AssemblyName just for this scenario.

Open to alternatives though.